Automattic/jetpack
# WAF: Options Improvements

## Proposed changes:

* The wpcom settings API casts boolean options values to `1` and `0`. These values get saved to the database as strings. Providing these string values directly in our config object opens up client implementations to the mistake of not considering `"0"` will evaluate to `true`.
* This is currently the case in at least the Jetpack settings UI for the WAF, so ensuring that the API provides a more reliable type will assist in backwards compatibility efforts as we introduce the WAF settings to Calypso, and start potentially having the `"0"` option value when a particular WAF feature is disabled.

### Other information:


## Jetpack product discussion



N/A

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






* Proof read, double check tests pass.
* Toggle automatic and manual rules, monitor the option values and contents of `wp-content/jetpack-waf` to validate normal functionality.
